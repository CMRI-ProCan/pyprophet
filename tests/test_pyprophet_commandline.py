import os
import subprocess
import shutil
import sys

import pytest

__here__ = os.path.dirname(os.path.abspath(__file__))


def _record(stdout, regtest):
    for line in stdout.split("\n"):
        if line.startswith("INFO"):
            # remove data which is not constant over runs as time or process id:
            line = line.split(":", 4)[-1]
        if "NEEDED" in line:
            continue
        if "time needed" in line:
            continue
        print >> regtest, line


def _run_cmdline(cmdline):
    stdout = cmdline + "\n"
    try:
        stdout += subprocess.check_output(cmdline, shell=True,
                                          stderr=subprocess.STDOUT)
    except subprocess.CalledProcessError, e:
        print >> sys.stderr, e.output
        raise
    return stdout


def _run_pyprophet_to_learn_model(regtest, temp_folder, dump_result_files=False, with_probatilities=False):
    os.chdir(temp_folder)
    data_path = os.path.join(__here__, "test_data.txt")
    shutil.copy(data_path, temp_folder)
    if with_probatilities:
        stdout = _run_cmdline("pyprophet test_data.txt --random_seed=42 --compute.probabilities")
    else:
        stdout = _run_cmdline("pyprophet test_data.txt --random_seed=42")
    for f in ["test_data_summary_stat.csv", "test_data_full_stat.csv", "test_data_report.pdf",
              "test_data_scorer.bin",
              "test_data_cutoffs.txt",
              "test_data_svalues.txt",
              "test_data_qvalues.txt",
              "test_data_with_dscore.csv",
              "test_data_with_dscore_filtered.csv",
              "test_data_dscores_top_target_peaks.txt",
              "test_data_dscores_top_decoy_peaks.txt",
              ]:
        full_path = os.path.join(temp_folder, f)
        assert os.path.exists(full_path)

    if not dump_result_files:
        return stdout

    for f in ["test_data_summary_stat.csv", "test_data_full_stat.csv",
              "test_data_cutoffs.txt",
              "test_data_svalues.txt",
              "test_data_qvalues.txt",
              "test_data_dscores_top_target_peaks.txt",
              "test_data_dscores_top_decoy_peaks.txt",
              "test_data_with_dscore.csv",
              "test_data_with_dscore_filtered.csv",
              ]:
        full_path = os.path.join(temp_folder, f)
        dump(full_path, regtest)

    full_path = os.path.join(temp_folder, "test_data_scorer.bin")
    dump_digest(regtest, full_path, "test_data_scorer.bin")

    return stdout

def dump(full_path, regtest):
    lines = open(full_path, "r").readlines()
    f = os.path.basename(full_path)
    print >> regtest
    print >> regtest, f, "contains", len(lines), "lines"
    print >> regtest
    if len(lines) > 10:
        print >> regtest, "top 5 lines of", f
        for line in lines[:5]:
            if len(line) > 80:
                line = line[:40] + " ... " + line[-45:]
            print >> regtest, line.rstrip()
        print >> regtest
        print >> regtest, "last 5 lines of", f
        for line in lines[-5:]:
            if len(line) > 80:
                line = line[:40] + " ... " + line[-45:]
            print >> regtest, line.rstrip()
    else:
        for line in lines:
            if len(line) > 80:
                line = line[:40] + " ... " + line[-45:]
            print >> regtest, line.rstrip()



def test_0(tmpdir, regtest):
    stdout = _run_pyprophet_to_learn_model(regtest, tmpdir.strpath, True, False)
    _record(stdout, regtest)


def test_1(tmpdir, regtest):
    stdout = _run_pyprophet_to_learn_model(regtest, tmpdir.strpath, True, True)
    _record(stdout, regtest)


def test_apply_classifier(tmpdir, regtest):

    stdout = _run_pyprophet_to_learn_model(regtest, tmpdir.strpath, True)

    _record(stdout, regtest)

    full_path = os.path.join(tmpdir.strpath, "test_data_with_dscore.csv")
    dump(full_path, regtest)

    full_path = os.path.join(tmpdir.strpath, "test_data_with_dscore_filtered.csv")
    dump(full_path, regtest)

    # collect m score stats
    m_score_stat, = [l for l in stdout.split("\n") if "mean m_score" in l]

    # split away log time etc:
    __, __, interesting_m_score_output = m_score_stat.partition("mean m_score")

    # collect s value stats
    s_value_stat, = [l for l in stdout.split("\n") if "mean s_value" in l]

    # split away log time etc:
    __, __, interesting_s_value_output = s_value_stat.partition("mean s_value")

    stdout = _run_cmdline("pyprophet test_data.txt --apply_scorer=test_data_scorer.bin "
                          "--target.overwrite --random_seed=42")

    _record(stdout, regtest)

    # collect m score stats
    m_score_stat, = [l for l in stdout.split("\n") if "mean m_score" in l]

    # split away log time etc:
    __, __, interesting_m_score_output2 = m_score_stat.partition("mean m_score")

    # collect s value stats
    s_value_stat, = [l for l in stdout.split("\n") if "mean s_value" in l]

    # split away log time etc:
    __, __, interesting_s_value_output2 = s_value_stat.partition("mean s_value")

    # check if output is consistent over runs:
    assert interesting_m_score_output == interesting_m_score_output2
    assert interesting_s_value_output == interesting_s_value_output2


def test_apply_weights(tmpdir, regtest):

    stdout = _run_pyprophet_to_learn_model(regtest, tmpdir.strpath, True)
    _record(stdout, regtest)

    stdout = _run_cmdline("pyprophet test_data.txt --apply_weights=test_data_weights.txt "
                          "--target.overwrite --random_seed=42")

    _record(stdout, regtest)

    for name in ["test_data_summary_stat.csv",
                 "test_data_full_stat.csv",
                 "test_data_report.pdf",
                 "test_data_cutoffs.txt",
                 "test_data_svalues.txt",
                 "test_data_qvalues.txt",
                 "test_data_dscores_top_target_peaks.txt",
                 "test_data_dscores_top_decoy_peaks.txt",
                 "test_data_with_dscore.csv",
                 "test_data_with_dscore_filtered.csv", ]:

        full_path = os.path.join(tmpdir.strpath, name)
        dump(full_path, regtest)

    full_path = os.path.join(tmpdir.strpath, "test_data_scorer.bin")
    dump_digest(regtest, full_path, "test_data_scorer.bin")


def dump_digest(regtest, full_path, name):
    import hashlib
    h = hashlib.sha1()
    h.update(open(full_path, "rb").read())
    print >> regtest
    print >> regtest, "digest", name, ":", h.hexdigest()



def test_multiple_input_files(tmpdir, regtest):
    os.chdir(tmpdir.strpath)

    data_path = os.path.join(__here__, "test_data_3.txt")
    shutil.copy(data_path, tmpdir.strpath)
    data_path = os.path.join(__here__, "test_data_2.txt")
    shutil.copy(data_path, tmpdir.strpath)

    stdout = _run_cmdline("pyprophet test_data_2.txt test_data_3.txt --random_seed=42 --target.overwrite")
    _record(stdout, regtest)


def test_multiple_input_files_with_merge(tmpdir, regtest):
    os.chdir(tmpdir.strpath)

    data_path = os.path.join(__here__, "test_data_3.txt")
    shutil.copy(data_path, tmpdir.strpath)
    data_path = os.path.join(__here__, "test_data_2.txt")
    shutil.copy(data_path, tmpdir.strpath)

    stdout = _run_cmdline("pyprophet test_data_2.txt test_data_3.txt --random_seed=42 --target.overwrite "
                          "--multiple_files.merge_results --compute.probabilities")

    _record(stdout, regtest)


def test_out_of_core_multi_input_files(tmpdir, regtest):

    def setup(subfolder):
        f = tmpdir.join(subfolder).strpath
        os.makedirs(f)
        os.chdir(f)
        data_path = os.path.join(__here__, "test_data_3.txt")
        shutil.copy(data_path, f)
        data_path = os.path.join(__here__, "test_data_2.txt")
        shutil.copy(data_path, f)
        return f

    f1 = setup("out_of_core_1.0")
    stdout = _run_cmdline("pyprophet test_data_2.txt test_data_3.txt "
                          "--out_of_core "
                          "--out_of_core.sampling_rate=1.0 --random_seed=42")
    _record(stdout, regtest)

    f1merge = setup("out_of_core_1.0_merged")
    stdout = _run_cmdline("pyprophet test_data_2.txt test_data_3.txt "
                          "--out_of_core "
                          "--out_of_core.sampling_rate=1.0 --random_seed=42 "
                          "--multiple_files.merge_results")

    _record(stdout, regtest)

    f2 = setup("in_core")
    stdout = _run_cmdline("pyprophet test_data_2.txt test_data_3.txt --random_seed=42")
    _record(stdout, regtest)

    f2merge = setup("in_core_merged")
    stdout = _run_cmdline("pyprophet test_data_2.txt test_data_3.txt --random_seed=42 "
                          "--multiple_files.merge_results")
    _record(stdout, regtest)

    f3 = setup("out_of_core_0.1")
    stdout = _run_cmdline("pyprophet test_data_2.txt test_data_3.txt "
                          "--out_of_core --out_of_core.sampling_rate=0.9999999 --random_seed=42")
    _record(stdout, regtest)

    print >> regtest
    print >> regtest, "the next three should be identical:"
    full_path = os.path.join(f1, "test_data__summary_stat.csv")
    dump(full_path, regtest)
    full_path = os.path.join(f1merge, "test_data__summary_stat.csv")
    dump(full_path, regtest)
    full_path = os.path.join(f2, "test_data__summary_stat.csv")
    dump(full_path, regtest)
    print >> regtest
    print >> regtest, "this might be different:"
    full_path = os.path.join(f3, "test_data__summary_stat.csv")
    dump(full_path, regtest)

    compare_folders(f1, f2)
    compare_folders(f1merge, f2merge)


def compare_folders(f1, f2):

    f1files = sorted([os.path.basename(fi) for fi in os.listdir(f1)])
    f2files = sorted([os.path.basename(fi) for fi in os.listdir(f2)])

    assert f1files == f2files

    for name in f1files:
        lines1 = open(os.path.join(f1, name), "r").readlines()
        lines2 = open(os.path.join(f1, name), "r").readlines()
        n1 = len(lines1)
        n2 = len(lines2)
        assert len(lines1) == len(lines2), (name, n1, n2)
        for i, (l1, l2) in enumerate(zip(lines1, lines2)):
            assert l1 == l2, (name, i, l1, l2)
