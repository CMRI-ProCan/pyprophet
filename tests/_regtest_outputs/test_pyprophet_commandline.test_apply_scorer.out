
test_data_summary_stat.csv contains 10 lines

qvalue	svalue	TP	FP	TN	FN	FDR	sens	cutoff
0.0	0.00223048327137	1.0	0.0	27.40414507 ... 837987253e-12	0.00278089968589	7.22470608404
0.01	1.0	360.0	3.0	24.4041450777	-0.4041 ... 414507772	0.00890555610518	1.0	1.67986704158
0.02	1.0	360.0	7.0	20.4041450777	-0.4041 ... 0414507772	0.0180909391855	1.0	1.04489819443
0.05	1.0	359.0	19.0	8.40414507772	0.5958 ... 0497358822056	0.998342987234	-0.363088375518
0.1	1.0	361.0	26.0	1.40414507772	-1.4041 ... 414507772	0.0681385402261	1.0	-1.58573235223
0.2
0.3
0.4
0.5

test_data_full_stat.csv contains 52 lines

top 5 lines of test_data_full_stat.csv
qvalue	svalue	TP	FP	TN	FN	FDR	sens	cutoff
0.0681385402261	1.0	361.0	26.0	1.4041450 ... 414507772	0.0681385402261	1.0	-2.02625427404
0.0681385402261	1.0	361.0	26.0	1.4041450 ... 414507772	0.0681385402261	1.0	-1.83242462844
0.0681385402261	1.0	361.0	26.0	1.4041450 ... 414507772	0.0681385402261	1.0	-1.63859498284
0.0681385402261	1.0	361.0	26.0	1.4041450 ... 414507772	0.0681385402261	1.0	-1.44476533725

last 5 lines of test_data_full_stat.csv
3.14418745669e-11	0.0356877323409	13.0	0 ... 4418745669e-11	0.0361516959165	6.88990942346
8.68656998184e-12	0.0133828996281	5.0	0. ... 9267055055e-12	0.0139044984294	7.08373906906
2.48837987253e-12	0.00223048327137	1.0	0 ... 837987253e-12	0.00278089968589	7.27756871466
2.48837987253e-12	0.00223048327137	1.0	0 ... 837987253e-12	0.00278089968589	7.47139836025
2.48837987253e-12	0.00223048327137	1.0	0 ... 837987253e-12	0.00278089968589	7.66522800585

test_data_report.pdf contains 544 lines

top 5 lines of test_data_report.pdf
%PDF-1.4
%¬Ü «º
1 0 obj
<< /Type /Catalog /Pages 2 0 R >>
endobj

last 5 lines of test_data_report.pdf
trailer
<< /Info 65 0 R /Root 1 0 R /Size 66 >>
startxref
25598
%%EOF

test_data_cutoffs.txt contains 1 lines

-2.026254e+00 -1.832425e+00 -1.638595e+0 ... 39e+00 7.277569e+00 7.471398e+00 7.665228e+00

test_data_svalues.txt contains 1 lines

1.000000e+00 1.000000e+00 1.000000e+00 1 ... 90e-02 2.230483e-03 2.230483e-03 2.230483e-03

test_data_qvalues.txt contains 1 lines

6.813854e-02 6.813854e-02 6.813854e-02 6 ... 70e-12 2.488380e-12 2.488380e-12 2.488380e-12

test_data_dscores_top_target_peaks.txt contains 1 lines

5.508691e+00 4.766004e+00 4.248958e+00 5 ... 77e+00 5.901192e+00 6.668852e+00 5.789941e+00

test_data_dscores_top_decoy_peaks.txt contains 1 lines

8.173490e-01 2.051536e+00 -5.752776e-03  ... 2e+00 -2.005057e-01 2.042997e-01 1.270100e+00

test_data_with_dscore.csv contains 9166 lines

top 5 lines of test_data_with_dscore.csv
transition_group_id	run_id	decoy	main_va ... series_score	d_score	m_score	peak_group_rank
459_run0	0	0	5.27893963303	5	0.993098149 ... 79341002	6	5.50869148316	1.09361806485e-08	1
459_run0	0	0	-0.13974375122	0	0.95392456 ... 6396888252	0	-1.3036449724	0.0679288328526	3
459_run0	0	0	-0.77504285962	0	0.66666624 ... 915833201	0	-4.72917362951	0.0681385402261	7
459_run0	0	0	-1.05454511848	0	0.84734030 ... 0811834697	0	-2.7911473241	0.0681385402261	4

last 5 lines of test_data_with_dscore.csv
DECOY_592_run0	0	1	-1.84945466301	0	0.63 ... 880632506	0	-2.83036473183	0.0681385402261	9
DECOY_592_run0	0	1	-2.10957129817	0	0.90 ... 335674837	0	-2.75949472559	0.0681385402261	8
DECOY_592_run0	0	1	-2.31560973872	0	0.90 ... 66613662	0	-4.25948498965	0.0681385402261	15
DECOY_592_run0	0	1	-2.53264045217	0	0.98 ... 96452491	0	-4.12553916112	0.0681385402261	14
DECOY_592_run0	0	1	-3.44062881815	0	-1.0 ... 	0.0	0.0	0	-3.57327980582	0.0681385402261	12

test_data_with_dscore_filtered.csv contains 9166 lines

top 5 lines of test_data_with_dscore_filtered.csv
transition_group_id	run_id	decoy	main_va ... series_score	d_score	m_score	peak_group_rank
459_run0	0	0	5.27893963303	5	0.993098149 ... 79341002	6	5.50869148316	1.09361806485e-08	1
459_run0	0	0	-0.13974375122	0	0.95392456 ... 6396888252	0	-1.3036449724	0.0679288328526	3
459_run0	0	0	-0.77504285962	0	0.66666624 ... 915833201	0	-4.72917362951	0.0681385402261	7
459_run0	0	0	-1.05454511848	0	0.84734030 ... 0811834697	0	-2.7911473241	0.0681385402261	4

last 5 lines of test_data_with_dscore_filtered.csv
DECOY_592_run0	0	1	-1.84945466301	0	0.63 ... 880632506	0	-2.83036473183	0.0681385402261	9
DECOY_592_run0	0	1	-2.10957129817	0	0.90 ... 335674837	0	-2.75949472559	0.0681385402261	8
DECOY_592_run0	0	1	-2.31560973872	0	0.90 ... 66613662	0	-4.25948498965	0.0681385402261	15
DECOY_592_run0	0	1	-2.53264045217	0	0.98 ... 96452491	0	-4.12553916112	0.0681385402261	14
DECOY_592_run0	0	1	-3.44062881815	0	-1.0 ... 	0.0	0.0	0	-3.57327980582	0.0681385402261	12
hex digtest pickled classifier: 3401eea774d1c7b2291022d662a56ef8fc53c7ae
pyprophet test_data.txt --apply_scorer=test_data_scorer.bin --target.overwrite --random_seed=42
 config settings:
     apply_scorer: test_data_scorer.bin
     apply_weights: None
     compute.probabilities: False
     d_score.cutoff: -1000.0
     delim.in:
     delim.out:
     export.mayu: False
     final_statistics.fdr_all_pg: False
     final_statistics.lambda: 0.4
     ignore.invalid_score_columns: 0
     is_test: 0
     multiple_files.merge_results: 0
     num_processes: 1
     out_of_core: 0
     out_of_core.sampling_rate: 0.1
     random_seed: 42
     semi_supervised_learner.initial_fdr: 0.15
     semi_supervised_learner.initial_lambda: 0.4
     semi_supervised_learner.iteration_fdr: 0.02
     semi_supervised_learner.iteration_lambda: 0.4
     semi_supervised_learner.num_iter: 5
     target.dir: None
     target.overwrite: 1
     target.prefix: None
     xeval.fraction: 0.5
     xeval.num_iter: 5
 apply scorer to input data
 processing input data finished
 data set contains 387 decoy and 387 target transition groups
 mean m_score = 6.130438e-02, std_dev m_score = 1.658378e-02
 mean s_value = 9.800913e-01, std_dev s_value = 1.125709e-01

used parameters:

    apply_scorer                               : 'test_data_scorer.bin'
    apply_weights                              : None
    compute.probabilities                      : False
    d_score.cutoff                             : -1000.0
    delim.in                                   : '\t'
    delim.out                                  : '\t'
    export.mayu                                : False
    final_statistics.fdr_all_pg                : False
    final_statistics.lambda                    : 0.4
    ignore.invalid_score_columns               : 0
    is_test                                    : 0
    multiple_files.merge_results               : 0
    num_processes                              : 1
    out_of_core                                : 0
    out_of_core.sampling_rate                  : 0.1
    random_seed                                : 42
    semi_supervised_learner.initial_fdr        : 0.15
    semi_supervised_learner.initial_lambda     : 0.4
    semi_supervised_learner.iteration_fdr      : 0.02
    semi_supervised_learner.iteration_lambda   : 0.4
    semi_supervised_learner.num_iter           : 5
    target.dir                                 : None
    target.overwrite                           : 1
    target.prefix                              : None
    xeval.fraction                             : 0.5
    xeval.num_iter                             : 5


WRITTEN:  test_data_with_dscore.csv
WRITTEN:  test_data_with_dscore_filtered.csv



test_data_with_dscore.csv contains 9166 lines

top 5 lines of test_data_with_dscore.csv
transition_group_id	run_id	decoy	main_va ... series_score	d_score	m_score	peak_group_rank
459_run0	0	0	5.27893963303	5	0.993098149 ... 79341002	6	5.50869148316	1.09361806485e-08	1
459_run0	0	0	-0.13974375122	0	0.95392456 ... 6396888252	0	-1.3036449724	0.0679288328526	3
459_run0	0	0	-0.77504285962	0	0.66666624 ... 915833201	0	-4.72917362951	0.0681385402261	7
459_run0	0	0	-1.05454511848	0	0.84734030 ... 0811834697	0	-2.7911473241	0.0681385402261	4

last 5 lines of test_data_with_dscore.csv
DECOY_592_run0	0	1	-1.84945466301	0	0.63 ... 880632506	0	-2.83036473183	0.0681385402261	9
DECOY_592_run0	0	1	-2.10957129817	0	0.90 ... 335674837	0	-2.75949472559	0.0681385402261	8
DECOY_592_run0	0	1	-2.31560973872	0	0.90 ... 66613662	0	-4.25948498965	0.0681385402261	15
DECOY_592_run0	0	1	-2.53264045217	0	0.98 ... 96452491	0	-4.12553916112	0.0681385402261	14
DECOY_592_run0	0	1	-3.44062881815	0	-1.0 ... 	0.0	0.0	0	-3.57327980582	0.0681385402261	12

test_data_with_dscore_filtered.csv contains 9166 lines

top 5 lines of test_data_with_dscore_filtered.csv
transition_group_id	run_id	decoy	main_va ... series_score	d_score	m_score	peak_group_rank
459_run0	0	0	5.27893963303	5	0.993098149 ... 79341002	6	5.50869148316	1.09361806485e-08	1
459_run0	0	0	-0.13974375122	0	0.95392456 ... 6396888252	0	-1.3036449724	0.0679288328526	3
459_run0	0	0	-0.77504285962	0	0.66666624 ... 915833201	0	-4.72917362951	0.0681385402261	7
459_run0	0	0	-1.05454511848	0	0.84734030 ... 0811834697	0	-2.7911473241	0.0681385402261	4

last 5 lines of test_data_with_dscore_filtered.csv
DECOY_592_run0	0	1	-1.84945466301	0	0.63 ... 880632506	0	-2.83036473183	0.0681385402261	9
DECOY_592_run0	0	1	-2.10957129817	0	0.90 ... 335674837	0	-2.75949472559	0.0681385402261	8
DECOY_592_run0	0	1	-2.31560973872	0	0.90 ... 66613662	0	-4.25948498965	0.0681385402261	15
DECOY_592_run0	0	1	-2.53264045217	0	0.98 ... 96452491	0	-4.12553916112	0.0681385402261	14
DECOY_592_run0	0	1	-3.44062881815	0	-1.0 ... 	0.0	0.0	0	-3.57327980582	0.0681385402261	12
pyprophet test_data.txt --out_of_core --apply_scorer=test_data_scorer.bin --target.overwrite --random_seed=42
 out_of_core setting ignored: this parameter has no influence for applying a persisted scorer
 config settings:
     apply_scorer: test_data_scorer.bin
     apply_weights: None
     compute.probabilities: False
     d_score.cutoff: -1000.0
     delim.in:
     delim.out:
     export.mayu: False
     final_statistics.fdr_all_pg: False
     final_statistics.lambda: 0.4
     ignore.invalid_score_columns: 0
     is_test: 0
     multiple_files.merge_results: 0
     num_processes: 1
     out_of_core: 1
     out_of_core.sampling_rate: 0.1
     random_seed: 42
     semi_supervised_learner.initial_fdr: 0.15
     semi_supervised_learner.initial_lambda: 0.4
     semi_supervised_learner.iteration_fdr: 0.02
     semi_supervised_learner.iteration_lambda: 0.4
     semi_supervised_learner.num_iter: 5
     target.dir: None
     target.overwrite: 1
     target.prefix: None
     xeval.fraction: 0.5
     xeval.num_iter: 5
 apply scorer to input data
 processing input data finished
 data set contains 387 decoy and 387 target transition groups
 mean m_score = 6.130438e-02, std_dev m_score = 1.658378e-02
 mean s_value = 9.800913e-01, std_dev s_value = 1.125709e-01

used parameters:

    apply_scorer                               : 'test_data_scorer.bin'
    apply_weights                              : None
    compute.probabilities                      : False
    d_score.cutoff                             : -1000.0
    delim.in                                   : '\t'
    delim.out                                  : '\t'
    export.mayu                                : False
    final_statistics.fdr_all_pg                : False
    final_statistics.lambda                    : 0.4
    ignore.invalid_score_columns               : 0
    is_test                                    : 0
    multiple_files.merge_results               : 0
    num_processes                              : 1
    out_of_core                                : 1
    out_of_core.sampling_rate                  : 0.1
    random_seed                                : 42
    semi_supervised_learner.initial_fdr        : 0.15
    semi_supervised_learner.initial_lambda     : 0.4
    semi_supervised_learner.iteration_fdr      : 0.02
    semi_supervised_learner.iteration_lambda   : 0.4
    semi_supervised_learner.num_iter           : 5
    target.dir                                 : None
    target.overwrite                           : 1
    target.prefix                              : None
    xeval.fraction                             : 0.5
    xeval.num_iter                             : 5


WRITTEN:  test_data_with_dscore.csv
WRITTEN:  test_data_with_dscore_filtered.csv



test_data_with_dscore.csv contains 9166 lines

top 5 lines of test_data_with_dscore.csv
transition_group_id	run_id	decoy	main_va ... series_score	d_score	m_score	peak_group_rank
459_run0	0	0	5.27893963303	5	0.993098149 ... 79341002	6	5.50869148316	1.09361806485e-08	1
459_run0	0	0	-0.13974375122	0	0.95392456 ... 6396888252	0	-1.3036449724	0.0679288328526	3
459_run0	0	0	-0.77504285962	0	0.66666624 ... 915833201	0	-4.72917362951	0.0681385402261	7
459_run0	0	0	-1.05454511848	0	0.84734030 ... 0811834697	0	-2.7911473241	0.0681385402261	4

last 5 lines of test_data_with_dscore.csv
DECOY_592_run0	0	1	-1.84945466301	0	0.63 ... 880632506	0	-2.83036473183	0.0681385402261	9
DECOY_592_run0	0	1	-2.10957129817	0	0.90 ... 335674837	0	-2.75949472559	0.0681385402261	8
DECOY_592_run0	0	1	-2.31560973872	0	0.90 ... 66613662	0	-4.25948498965	0.0681385402261	15
DECOY_592_run0	0	1	-2.53264045217	0	0.98 ... 96452491	0	-4.12553916112	0.0681385402261	14
DECOY_592_run0	0	1	-3.44062881815	0	-1.0 ... 	0.0	0.0	0	-3.57327980582	0.0681385402261	12

test_data_with_dscore_filtered.csv contains 9166 lines

top 5 lines of test_data_with_dscore_filtered.csv
transition_group_id	run_id	decoy	main_va ... series_score	d_score	m_score	peak_group_rank
459_run0	0	0	5.27893963303	5	0.993098149 ... 79341002	6	5.50869148316	1.09361806485e-08	1
459_run0	0	0	-0.13974375122	0	0.95392456 ... 6396888252	0	-1.3036449724	0.0679288328526	3
459_run0	0	0	-0.77504285962	0	0.66666624 ... 915833201	0	-4.72917362951	0.0681385402261	7
459_run0	0	0	-1.05454511848	0	0.84734030 ... 0811834697	0	-2.7911473241	0.0681385402261	4

last 5 lines of test_data_with_dscore_filtered.csv
DECOY_592_run0	0	1	-1.84945466301	0	0.63 ... 880632506	0	-2.83036473183	0.0681385402261	9
DECOY_592_run0	0	1	-2.10957129817	0	0.90 ... 335674837	0	-2.75949472559	0.0681385402261	8
DECOY_592_run0	0	1	-2.31560973872	0	0.90 ... 66613662	0	-4.25948498965	0.0681385402261	15
DECOY_592_run0	0	1	-2.53264045217	0	0.98 ... 96452491	0	-4.12553916112	0.0681385402261	14
DECOY_592_run0	0	1	-3.44062881815	0	-1.0 ... 	0.0	0.0	0	-3.57327980582	0.0681385402261	12
